clients/admin-ui/src/features/system/SystemFormTabs.tsx

```diff
@@ -0,0 +1,208 @@
+        </Button>
+        .
+      </Text>
+    </Box>
+  );
+};
+
+const SystemFormTabs = ({
+  isCreate,
+}: {
+  /** If true, then some editing features will not be enabled */
+  isCreate?: boolean;
+}) => {
+  const [tabIndex, setTabIndex] = useState(0);
+  const [queuedIndex, setQueuedIndex] = useState<number | undefined>(undefined);
+  const [showSaveMessage, setShowSaveMessage] = useState(false);
+  const { systemOrDatamapRoute } = useInterzoneNav();
+  const router = useRouter();
+  const toast = useToast();
+  const dispatch = useAppDispatch();
+  const activeSystem = useAppSelector(selectActiveSystem);
+
+  const goBack = useCallback(() => {
+    router.back();
+    dispatch(setActiveSystem(undefined));
+  }, [dispatch, router]);
+
+  const handleSuccess = (system: System) => {
+    // show a save message if this is the first time the system was saved
+    if (activeSystem === undefined) {
+      setShowSaveMessage(true);
+    }
+    dispatch(setActiveSystem(system));
+    const toastParams = {
+      ...DEFAULT_TOAST_PARAMS,
+      description: (
+        <ToastMessage
+          onViewDatamap={() => {
+            router.push(systemOrDatamapRoute).then(() => {
+              toast.closeAll();
+            });
+          }}
+          onAddPrivacyDeclaration={() => {
+            setTabIndex(1);
+            toast.closeAll();
+          }}
+        />
+      ),
+    };
+    toast({ ...toastParams });
+  };
+
+  useEffect(() => {
+    /**
+     * The first time this component mounts, if it's a create, make sure we don't have an active system
+     * This can happen if the user was editing a system, then navigated away by typing in a new URL path
+     * When navigating not through a URL path, the return unmount should handle resetting the system
+     */
+    if (isCreate) {
+      dispatch(setActiveSystem(undefined));
+    }
+    return () => {
+      // on unmount, unset the active system
+      dispatch(setActiveSystem(undefined));
+    };
+  }, [dispatch, isCreate]);
+
+  const checkTabChange = (index: number) => {
+    // While privacy declarations aren't updated yet, only apply the "unsaved changes" modal logic
+    // to the system information tab
+    if (index === 0) {
+      setTabIndex(index);
+    } else {
+      setQueuedIndex(index);
+    }
+  };
+
+  const continueTabChange = () => {
+    if (queuedIndex) {
+      setTabIndex(queuedIndex);
+      setQueuedIndex(undefined);
+    }
+  };
+
+  const tabData: TabData[] = [
+    {
+      label: "System information",
+      content: (
+        <>
+          <Box px={6} mb={9}>
+            <SystemInformationForm
+              onSuccess={handleSuccess}
+              system={activeSystem}
+              abridged={isCreate}
+            >
+              <UnmountWarning
+                isUnmounting={queuedIndex !== undefined}
+                onContinue={continueTabChange}
+                onCancel={() => setQueuedIndex(undefined)}
+              />
```

Comment:
tbh I don't love this pattern, but I think it works decently well. the problem with it is I'd rather have the unmount be observed by the tab component, since now every component that is part of the tab either needs this `UnmountWarning` component as a child, or we need special logic to go around the queue/step logic (like the `if` statement in `checkTabChange`. the UnmountWarning needs to be a child in order to have access to the formik context to see if the form is dirty.

the experience team also ran into this recently and [handled it through refs](https://github.com/ethyca/fides/blob/main/clients/admin-ui/src/features/common/custom-fields/CustomFieldsModal.tsx#L49-L62) which I didn't love, but maybe I'm just averse to refs

Reply:
This is definitely a tricky issue. I found a potential bug with the unsaved changes logic. If it would be too complex or time consuming then I don't think it's worth fixing in this PR. It seems like a good follow up bug.

The issue is if a user clicks on a link to navigate away instead of the privacy declaration tab it doesn't prompt them to save their changes.

https://user-images.githubusercontent.com/17103888/219412682-b1232477-a2a3-45fc-a61b-caa0f46b28a5.mov


Maybe in the future we could standard on a solution that we like? It might confusing if there are multiple ways that we're handling this scenario

Reply:
yeah, I couldn't figure out a way to render the modal while the component is unmounting (which is what happens when you click away). it also wouldn't work if the user actually entered a different URL to navigate away (the component don't go through their unmounting phase that way).

actually maybe I'm going about this wrong and should instead try to use next router to see if I can hook into that... that wouldn't work for if the user actually navigated away from the page (going from localhost:3000 --> i.e. google.com), but hey we can't account for _everything_

I'll try it out!

Reply:
okay the good news is that we can probably intercept next router and window events to handle this

the bad news is that it's fairly complicated 😅 

https://github.com/vercel/next.js/discussions/12348

I'm going to make a separate ticket for this since otherwise I'll spend too much time on it right now

Reply:
https://github.com/ethyca/fides/issues/2631